### ISSUE TITLE
Fixing OPTIONS response headers

### ISSUE THREAD

**Issue by EmmanVazz** [LearningLocker/learninglocker#1236]
**Version**
v2.5.0

**Steps to reproduce the bug**
1. Send OPTIONS request to activities state (ex 
http://localhost:3000/data/xAPI/activities/state?stateId=resume&activityId=http%3A%2F%2F6RqRmZwBzUE_course_id&agent=%7B%22objectType%22%3A%22Agent%22%2C%22mbox%22%3A%22mailto%3Afirstlast%40mycompany.com%22%2C%22name%22%3A%22First%20Last%22%7D)

2. Course failing to start due to missing Access-Control-Allow headers in response

**Expected behaviour**
Match the cloud.scorm headers
![snip20180823_6](https://user-images.githubusercontent.com/1174973/44563991-75197a80-a726-11e8-9055-87c402c68a31.png)

**Actual behaviour**
![snip20180823_4](https://user-images.githubusercontent.com/1174973/44563998-7c408880-a726-11e8-9b9c-e3c7fa96b3a6.png)

**Additional information**
The articulate tin can course makes this options request to the LRS. However due to the LRS not telling the course what methods, headers or origins it accepts it can't start. 

I have seen [this issue](https://github.com/LearningLocker/learninglocker/issues/754) and tried to follow along but it seems like all the information is for v1 of this project and no longer relevant.

**Comment by ht2** [comment]
We fully implement the npm cors package, I'm pretty sure this is an issue from attemtpting to access CORS via localhost. 

Try it on a properly resolved host and see if you still have the same issue.

**Comment by EmmanVazz** [comment]
Hmm found the config [here](https://github.com/LearningLocker/learninglocker/blob/12892111d74a27c473891b36b2b8eef2926ba788/api/src/server.js#L15). However it doesn't seem to change the results for the x-api calls. Looked for a similar congif setup in that project but don't see it or is there some command I have to run for it to take the new config? I am using pm2 to run it.

**Comment by EmmanVazz** [comment]
Hmm it seems like it is only taking in the origin setting, it is ignoring the methods and allowedHeaders settings I am giving it.

**Comment by EmmanVazz** [comment]
Well this is a workaround for the config not being take in. https://stackoverflow.com/a/46051141/4880012 However still not sure how to get that to work with the x-api calls.

**Comment by EmmanVazz** [comment]
I have added that code from the stack overflow answer [here](https://github.com/LearningLocker/xapi-service/blob/08785f99ea9fa979878719d270ac7b05fab82986/src/server.ts#L42) in the x-api project. Closing issue. Thanks for the help.

**Comment by ht2** [comment]
OK, good stuff. For the record (and anyone else looking) we import the CORS mixin from our JS commons repo here: https://github.com/LearningLocker/xapi-statements/blob/master/src/expressPresenter/index.ts#L33

Mixin can be found here: https://github.com/HT2-Labs/jscommons/blob/master/src/expressPresenter/mixins/cors.ts

**Comment by EmmanVazz** [comment]
lol yeah that is a clear solution.

**Comment by EmmanVazz** [comment]
lmao closing this. data/xAPI goes to https://github.com/LearningLocker/learninglocker/blob/12892111d74a27c473891b36b2b8eef2926ba788/ui/src/server.js#L24 so place cors code there. Thanks closing again and for good this time.

**Comment by kang1happy** [comment]
in my opinion this issue was resolved, but I get this issue when I use url 'http://mytest/courselibarary/170/story_html5.html?tincan=true&actor={"name":["emai@email.com"],"account":[{"accountServiceHomePage":"http://mytest/lms/","accountName":"email@email.com"}],"objectType":"Agent"}&endpoint=http://server0001:3000/data/xAPI/&auth=Basic YjgyODFiYjc5MDcwYTgzOGU3MDM2ZTdiZmQ4NTJkODk5ZTBiNmFjNDozYjNkODA3MWU3YjI4M2FmM2ZhNzliYjQ0NDkxNDJkM2IzYzZiMTE1&activity_id =http://5VymcRv7eZm_course_id&registration=f6c527d5-8ff4-4436-a193-a82394c6f3b2' to review the course and submit result to learninglocker i get following issue from origin 'http://mytest' has been blocked by CORS policy: 
Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.
my learninglocker was the latest version v5.1.0, and use PM2 to manage the services
is there any config I was missing? anyone have any good suggestion, thanks a lot.

**Comment by kang1happy** [comment]
> in my opinion this issue was resolved, but I get this issue when I use url 'http://mytest/courselibarary/170/story_html5.html?tincan=true&actor={"name":["emai@email.com"],"account":[{"accountServiceHomePage":"http://mytest/lms/","accountName":"email@email.com"}],"objectType":"Agent"}&endpoint=http://server0001:3000/data/xAPI/&auth=Basic YjgyODFiYjc5MDcwYTgzOGU3MDM2ZTdiZmQ4NTJkODk5ZTBiNmFjNDozYjNkODA3MWU3YjI4M2FmM2ZhNzliYjQ0NDkxNDJkM2IzYzZiMTE1&activity_id =http://5VymcRv7eZm_course_id&registration=f6c527d5-8ff4-4436-a193-a82394c6f3b2' to review the course and submit result to learninglocker i get following issue from origin 'http://mytest' has been blocked by CORS policy:
> Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.
> my learninglocker was the latest version v5.1.0, and use PM2 to manage the services
> is there any config I was missing? anyone have any good suggestion, thanks a lot.

BTW, we just upgrade this version from V1，we have got the old data by UI, in the old version V1, we would like to use config by this solution 'https://stackoverflow.com/questions/7067966/why-doesnt-adding-cors-headers-to-an-options-route-allow-browsers-to-access-my/46051141#46051141', and it's work fine, so I'm not sure is there any config for V2+
